gym/envs/box2d/car_racing.py

```diff
@@ -81,6 +82,14 @@ def _contact(self, contact, begin):
                 tile.road_visited = True
                 self.env.reward += 1000.0 / len(self.env.track)
                 self.env.tile_visited_count += 1
+
+                # Lap is considered completed if enough % of the track was covered
+                if (
+                    tile.idx == 0
+                    and self.env.tile_visited_count / len(self.env.track)
+                    > self.lap_complete_percent
+                ):
+                    self.env_new_lap = True
```

Comment:
I just realized this should be `self.env.new_lap` no?
and in that case, termination is due to reaching lap end (should not be treated as normal termination)